### PR TITLE
Add tests for DiagnosticsJob

### PR DESCRIPTION
**Pull request by janisz**
Currenlty, the code has no tests. In this PR I added a bunch of tests for `DiagnosticsJob` and cleanup code a little bit.

### CODE REVIEW

File: api/diagnostics.go [hunk 1]
```diff
@@ -484,14 +484,13 @@ func (d diagnosticsJobCanceledError) Error() string {
 }
 
 // fetch an HTTP endpoint and append the output to a zip file.
-func (j *DiagnosticsJob) getHTTPAddToZip(node Node, endpoints map[string]string, folder string, zipWriter *zip.Writer,
```
Comment (philipnrmn) [inline api/diagnostics.go]: How did this ever build? I would expect errors on the unused parameter...

File: api/diagnostics.go [hunk 2]
```diff
@@ -518,32 +517,11 @@ func (j *DiagnosticsJob) getHTTPAddToZip(node Node, endpoints map[string]string,
...
-
-		if resp.StatusCode != http.StatusOK {
-			errMsg := fmt.Sprintf("unable to fetch %s. Return code %d", fullURL, resp.StatusCode)
-			logrus.Error(errMsg)
-			j.Errors = append(j.Errors, errMsg)
-			updateSummaryReport("wrong status code returned", node, errMsg, summaryErrorsReport)
+			logrus.Errorf("Could not read force-tls flag: %s", err)
+			updateSummaryReport(err.Error(), node, err.Error(), summaryErrorsReport)
 			j.JobProgressPercentage += percentPerURL
 			continue
```
Comment (philipnrmn) [inline api/diagnostics.go]: We don't close the body in this case; isn't this going to lead to hanging connections?

Comment (gpaul) [reply api/diagnostics.go]: If you close the response body here 
https://github.com/dcos/dcos-diagnostics/pull/60/files#diff-0ee9de7b9d9927cce9fb53251662d72aR564 (so if the request succeeded but the response is unexpected) then the `get()` function maintains a contract that says if it returns an error, the response body does not need to be closed. 

That's a convenient pattern to follow.

File: api/diagnostics.go (hunk 1 above)
Comment (janisz) [reply api/diagnostics.go]: Good point. The check was disabled. I fixed it here https://github.com/dcos/dcos-diagnostics/pull/62

File: api/diagnostics.go (hunk 2 above)
Comment (janisz) [reply api/diagnostics.go]: Fixed

**Comment by janisz** [comment]
@philipnrmn Fixed

File: api/diagnostics.go [hunk 3]
```diff
@@ -485,12 +485,11 @@ func (d diagnosticsJobCanceledError) Error() string {
 // fetch an HTTP endpoint and append the output to a zip file.
 func (j *DiagnosticsJob) getHTTPAddToZip(node Node, endpoints map[string]string, zipWriter *zip.Writer,
 	summaryErrorsReport, summaryReport *bytes.Buffer, percentPerNode float32) error {
-	if len(endpoints) == 0 || percentPerNode == 0 {
```
Comment (gracedo) [inline api/diagnostics.go]: just wondering why these checks can be removed? (we don't need to worry about a divide by 0 panic?)

File: api/diagnostics.go [hunk 4]
```diff
@@ -517,32 +516,11 @@ func (j *DiagnosticsJob) getHTTPAddToZip(node Node, endpoints map[string]string,
...
-			j.JobProgressPercentage += percentPerURL
-			continue
-		}
-
-		if resp.StatusCode != http.StatusOK {
-			errMsg := fmt.Sprintf("unable to fetch %s. Return code %d", fullURL, resp.StatusCode)
-			logrus.Error(errMsg)
-			j.Errors = append(j.Errors, errMsg)
-			updateSummaryReport("wrong status code returned", node, errMsg, summaryErrorsReport)
+			logrus.Errorf("Could not read force-tls flag: %s", err)
```
Comment (gracedo) [inline api/diagnostics.go]: this blanket error message doesn't look quite right

File: api/diagnostics.go (hunk 3 above)
Comment (janisz) [reply api/diagnostics.go]: It's covered here https://github.com/dcos/dcos-diagnostics/pull/60/files#diff-0ee9de7b9d9927cce9fb53251662d72aR303 (Github fold this part).

File: api/diagnostics.go (hunk 4 above)
Comment (janisz) [reply api/diagnostics.go]: Good point

File: api/diagnostics.go (hunk 3 above)
Comment (janisz) [reply api/diagnostics.go]: I mocked `/logs` endpoint in one tests to validate this case.

**Comment by janisz** [comment]
@philipnrmn @gracedo Can you look one more time?

**Pull request merged**